app/javascript/src/LoginPage/loginForms/SignupForm.jsx

```diff
@@ -0,0 +1,127 @@
+// @flow
+
+import React, { Component } from 'react'
+import {
+  Form,
+  ActionGroup
+} from '@patternfly/react-core'
+import {
+  HiddenInputs,
+  FormGroup
+} from 'LoginPage'
+
+type Props = {
+  path: string,
+  user: {
+    email: string,
+    username: string,
+    firstname: string,
+    lastname: string
+  }
+}
+
+type State = {
+  username: string,
+  emailAddress: string,
+  firstname: string,
+  lastname: string,
+  password: string,
+  passwordConfirmation: string,
+  isFilledUsername: ?boolean,
```

Comment:
I honestly think this is not needed at all, you could validate each field when it's updated (and/or submit) that it's not empty and do something about it (i.e: show error message). This is something that just pollutes the state.